integration/tools/validation/src/main/java/alluxio/cli/fuse/CorrectnessValidationOptions.java

```diff
@@ -0,0 +1,163 @@
+package alluxio.cli.fuse;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+/**
+ * This class stores the options passed in from CLI.
+ */
+public final class CorrectnessValidationOptions {
+
+  public static final CommandLineParser PARSER = new DefaultParser();
+  private final String mLocalDir;
+  private final String mFuseDir;
+  private final CorrectnessValidationOperation mOperation;
+  private final int mNumThreads;
+  private final int mNumFiles;
+
+  /**
+   * Constructs a {@CorrectnessValidationOptions} object through CLI.
+   * @param args arguments from CLI
+   * @return a {@CorrectnessValidationOptions} holding all the test options
+   * @throws ParseException
+   * @throws IllegalArgumentException
+   */
+  public static CorrectnessValidationOptions createOptions(String[] args)
+      throws ParseException, IllegalArgumentException {
+    CommandLine cli = PARSER.parse(CorrectnessOptionsParser.OPTIONS, args);
+    String localDir = cli.getOptionValue(CorrectnessOptionsParser.LOCAL_DIR_OPTION_NAME);
+    String fuseDir = cli.getOptionValue(CorrectnessOptionsParser.FUSE_DIR_OPTION_NAME);
+    CorrectnessValidationOperation operation = CorrectnessValidationOperation.fromString(
+        cli.getOptionValue(CorrectnessOptionsParser.TEST_OPERATION_OPTION_NAME));
+    int numThreads = Integer.parseInt(
+        cli.getOptionValue(CorrectnessOptionsParser.THREAD_NUMBER_OPTION_NAME));
+    int numFiles = Integer.parseInt(
+        cli.getOptionValue(CorrectnessOptionsParser.FILE_NUMBER_OPTION_NAME));
+
+    return new CorrectnessValidationOptions(localDir, fuseDir, operation, numThreads, numFiles);
+  }
+
+  private CorrectnessValidationOptions(String localDir, String fuseDir,
+      CorrectnessValidationOperation operation, int numThreads, int numFiles) {
+    Preconditions.checkNotNull(localDir, "Option localDir should not be null.");
+    Preconditions.checkNotNull(fuseDir, "Option fuseDir should not be null.");
+    Preconditions.checkNotNull(operation, "Option operation should not be null.");
+    Preconditions.checkNotNull(numThreads, "Option numThreads should not be null.");
+    Preconditions.checkNotNull(numFiles, "Option numFiles should not be null.");
+    validateOptions(operation, numThreads);
+    mLocalDir = localDir;
+    mFuseDir = fuseDir;
+    mOperation = operation;
+    mNumThreads = numThreads;
+    mNumFiles = numFiles;
+  }
+
+  private void validateOptions(CorrectnessValidationOperation operation, int numThreads) {
+    if (operation == CorrectnessValidationOperation.WRITE && numThreads != 1) {
+      System.out.println("AlluxioFuse only supports single thread writing.");
+      System.exit(1);
```

Comment:
System.exit() should only be done by the main thread in the main function.
FYI https://github.com/Alluxio/alluxio/pull/15699
the similar issue caused by my code

Here you should just throw an RuntimeException
and main() function can catch and do exit or exit with runtime exception directly
based on the user experience you prefer (user see well formatted output or an exception)

Reply:
Similar for all other places except main function